src/Microsoft.AspNetCore.Razor.Evolution/RazorSourceDocument.cs

```diff
@@ -17,6 +17,8 @@ public abstract class RazorSourceDocument
 
         public abstract int Length { get; }
 
+        public abstract RazorSourceLineCollection Lines { get; }
```

Comment:
Seems odd for this guy to be on the `RazorSourceDocument`.  Seems more suited for its methods to be exposed as extension methods. Aka `GetSourceLocation(this RazorSourceDocument document, int position)` etc.